src/FileConventions/Library.fs

```diff
@@ -0,0 +1,10 @@
+﻿module FileConventions
+
+open System
+open System.IO
+open System.Linq
+
+let HasBinaryContent (fileInfo: FileInfo) =
+    use streamReader = new StreamReader (fileInfo.FullName)
+    let content = streamReader.ReadToEnd()
+    content.Any(fun ch -> Char.IsControl(ch) && ch <> '\r' && ch <> '\n')

```

Comment:
@realmarv we already talked about how using .ReadToEnd() could have bad performance implications, let's use the same technique that you have learned before to prevent this, it's just 2 lines of code change (NOTE: I'm not expecting you now that, by default without asking me, you seek the algorithm with the best performance, because 99% of the time we don't need that; however if you have already learned how to prevent a performance pitfall that only takes a 2 LOC change and zero research, like in this case, please apply it without hesitation and without asking me.)

Reply:
@realmarv nit: do not use the abbreviation "ch", there's no point in saving characters in source code

Reply:
> @realmarv we already talked about how using .ReadToEnd() could have bad performance implications, let's use the same technique that you have learned before to prevent this, it's just 2 lines of code change (NOTE: I'm not expecting you now that, by default without asking me, you seek the algorithm with the best performance, because 99% of the time we don't need that; however if you have already learned how to prevent a performance pitfall that only takes a 2 LOC change and zero research, like in this case, please apply it without hesitation and without asking me.)

In the previous case, we only needed the first line of the code. but in this case, I'm not sure how many lines are enough. So, should I iterate over the lines of the file until I find a control character?

Reply:
> I'm not sure how many lines are enough

You're using Any(), so **all** lines need to be iterated. But it doesn't matter, because ReadLines() goes line by line, instead of putting the whole file in memory (imagine if you try to read a 1GB this way).

Reply:
Done